### DISCUSSION TITLE
Menu de clases

### DISCUSSION THREAD

**Discussion by Alejandrojuarez0105** [mmasias/23-24-prg1#172] [Show and tell]
Hola Masias, le comento que intente realizar un "menú" el cual te indique que debes estudiar (basado en nuestras clases) y me gustaría saber si habrá alguna manera de mejorarlo o que se vea más limpio. Link donde se encuentra el código: https://github.com/Alejandrojuarez0105/Repaso-progra-1/tree/master

**Comment by mmasias** [comment]
Hola Alejandro

Te recomiendo ir de a pocos:

Céntrate para empezar en el primer nivel de preguntas, generalizalo, de
modo que tengas un único método al que le pases deseoEstudioUsuario, una
matriz con las asignaturas y una matriz con los temas; y que sea ese método
sea el que le pregunte si desea elegir un reto o si lo desea aleatorio.

Las matrices de temas y asignaturas las puedes modelar por ejemplo así:

```java
String[] asignaturas = {"Progra1", "iGPySw", "TEO", "Mate", "Fisica"}

String[][] temasPorAsignatura = {
     {"Entrada", "Salida", "Variables", "Bucles", "Condicionales",
"Matrices"},
     {"Tema 1 iGPySw", "Tema 2 iGPySw", "Tema 3 iGPySw", "Tema 4 iGPySw",
"Tema 5 iGPySw", },
     {"Tema 1 TEO", "Tema 2 TEO", "Tema 3 TEO", "Tema 4 TEO", "Tema 5 TEO",
},
     {"Tema 1 Mate", "Tema 2 Mate", "Tema 3 Mate", "Tema 4 Mate", "Tema 5
Mate", },
     {"Tema 1 Fisica", "Tema 2 Fisica", "Tema 3 Fisica", "Tema 4 Fisica",
"Tema 5 Fisica", },
}
```

Entonces, si defines

```java
final int PROGRAMACION_1 = 0;
final int IGPYSW = 1;
...
final int FISICA = 4;
```

El nombre de la asignatura fisica lo puedes sacar usando:

```java
asignaturas[FISICA]
```

Y la matriz de temas de física sería:

```java
temasPorAsignatura[FISICA]
```

Es una idea de partida para que puedas darle vueltas.

Un saludo
